src/main/java/io/reactivex/internal/operators/maybe/MaybeTimer.java

```diff
@@ -38,40 +38,16 @@ public MaybeTimer(long delay, TimeUnit unit, Scheduler scheduler) {
     }
 
     @Override
-    protected void subscribeActual(MaybeObserver<? super Long> observer) {
-        TimerDisposable parent = new TimerDisposable(observer);
-        observer.onSubscribe(parent);
+    protected void subscribeActual(final MaybeObserver<? super Long> observer) {
+        SequentialDisposable sd = new SequentialDisposable();
 
-        parent.setFuture(scheduler.scheduleDirect(parent));
-    }
-
-    static final class TimerDisposable extends AtomicReference<Disposable> implements Disposable, Runnable {
-
-        /** */
-        private static final long serialVersionUID = 2875964065294031672L;
-        final MaybeObserver<? super Long> actual;
-
-        public TimerDisposable(MaybeObserver<? super Long> actual) {
-            this.actual = actual;
-        }
-
-        @Override
-        public void run() {
-            actual.onSuccess(0L);
-        }
-
-        @Override
-        public void dispose() {
-            DisposableHelper.dispose(this);
-        }
-
-        @Override
-        public boolean isDisposed() {
-            return DisposableHelper.isDisposed(get());
-        }
+        observer.onSubscribe(sd);
 
-        void setFuture(Disposable d) {
-            DisposableHelper.replace(this, d);
-        }
+        sd.replace(scheduler.scheduleDirect(new Runnable() {
```

Comment:
It was okay with the named class, we should be moving away from anonymous inner classes.